Add maintenance process kb

chloesoe opened the pull request:
A customer was asking about the vulnerability and patching concept at
APPUiO Public. We do not have written down our maintenance process yet.
The couple of sentences which were written for the customer are now
documented here and could be extended. Originally it was in our wiki in
VINT space with name "Maintenance Process".

Commit:
Add maintenance process kb

A customer was asking about the vulnerability and patching concept at
APPUiO Public. We do not have written down our maintenance process yet.
The couple of sentences which were written for the customer are now
documented here and could be extended. Originally it was in our wiki in
VINT space with name "Maintenance Process".

Commit:
Define updates on APPUiO public more precisely

Updates in the Openshift platforms, like bug fixes, are not installed
automatically during weekly maintenance. They are installed, if affected
managed platforms.

tobru commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,18 @@
+== Overview
+This page should provide information about our maintenance process.
+
+== APPUiO Public Maintenance Process
```
This sections belongs to the APPUiO docs under https://github.com/appuio/docs

chloesoe replied:
Hmm, had a look on https://github.com/appuio/docs and I'm not sure if it really fits there. In APPUiO docs we do have this main sections:
* User Documentation
* APPUiO Microservices Example
* Application Tutorials

The "User Documentation" would be the best fit, but it seems to me it is more a "self service" section, where a APPUiO customer could get information to run their service.

Where would it best fit in your opinion? Perhaps a paragraph in http://docs.appuio.ch/en/latest/appuio-specifics-public.html

So in my opinion it is a better fit in the general KB about VSHN. Maintenance is a service by VSHN. And if somebody wants to know what maintenance process we provide, then it should be on centralized place.
Please let me know what you think about that.

tobru replied:
We can leave the documentation in kb, but then a link from docs.appuio.ch to kb.vshn.net is needed, pointing to this page. http://docs.appuio.ch/en/latest/appuio-specifics-public.html would be a good place for it, yes.

tobru commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,18 @@
+= Maintenance Process
+
+== Overview
+This page should provide information about our maintenance process.
```
This page not only "should" provide, it actually does. Please rephrase.

tobru reviewed (changes requested):
Thanks. I'm not confident about the wording and information provided, it somehow lacks substance for someone who wants / needs to understand how our maintenance is done and what we actually do. Would you want to give this another try?

Commit:
Fix wording and typo

chloesoe commented:
> Thanks. I'm not confident about the wording and information provided, it somehow lacks substance for someone who wants / needs to understand how our maintenance is done and what we actually do. Would you want to give this another try?

It was a first shot when a customer asked because of a security review about how fast security updates for vulnerabilities on APPUiO are installed. Since we did not have a sentences ready to use I wrote this down for APPUiO. I will have a look on it to give more substance for the "Managed Server Maintenance Process"; there I actually should know how the process works, since I also do maintenance regularly. I'm not sure yet how deeply I should document the process.

Commit:
Rewrite the managed server maintenance completely

Since there was to less substance in the first commit, now the try to
add some more information. It is written in prose text as it is
experienced by the maintenance engineer.

Ticket: VINT-1581

Commit:
Add information about different Squads in overview

Ticket: VINT-1581

Commit:
Add information about SaaS/PaaS update

Ticket: VINT-1581

chloesoe commented:
@tobru: I completely rewrote the maintenance part of the managed server maintenance. See commits:
* 04128fdfb5b4d7e4aaeeabfef912ec6b58e6d7e4 (main rewrite)
* b001ca1b0ffa5d2d321c623237d36e1989530ba8 (additional info)
* 868e445539cffdfbee500c801d8e0ffaff65fff2 (additional info)

I am not sure if I fit the requirements. I mostly wrote in prose text how I experience the maintenance. Don't know if that style is liked on the handbook.

Probably there writing style could be improved. Perhaps @akosma could have a look on the text.

tobru commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+== Overview
+This page provides information about our maintenance process.
+Basically we have two kind of maintenances procedures.
+For the APPUiO clusters and Openshift clusters our internal Squad _Sirius_ is responsible.
```
I guess people reading the KB don't want to know about Squads. I propose to leave this paragraph out, as we provide maintenance "as VSHN" and it shouldn't make any difference which Squad does the work.

tobru reviewed (changes requested):
In general I like it very much now. Just some small adjustments, then we can merge it.

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+== APPUiO Public Maintenance Process
+APPUiO Public servers are updated every Tuesday with the current security and software patches provided by the Linux distributions.
+Critical security vulnerabilities in the Openshift software are announced by RedHat via https://access.redhat.com/security/security-updates/#/[Security Advisories], whose mailing list includes the VSHN Engineers.
+This security advisories are analysed within one business day.
```
(Grammar) -- I'd suggest:

> Security advisories are analysed within one business day.

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+
+== APPUiO Public Maintenance Process
+APPUiO Public servers are updated every Tuesday with the current security and software patches provided by the Linux distributions.
+Critical security vulnerabilities in the Openshift software are announced by RedHat via https://access.redhat.com/security/security-updates/#/[Security Advisories], whose mailing list includes the VSHN Engineers.
```
(Spelling) -- OpenShift
(Lowercase) -- VSHN engineers
(Wording) -- Instead of:

> , whose mailing list includes the VSHN Engineers.

I would suggest:

> and we're subscribed to their notifications.

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+APPUiO Public servers are updated every Tuesday with the current security and software patches provided by the Linux distributions.
+Critical security vulnerabilities in the Openshift software are announced by RedHat via https://access.redhat.com/security/security-updates/#/[Security Advisories], whose mailing list includes the VSHN Engineers.
+This security advisories are analysed within one business day.
+If a vulnerability affects the system directly, the update is planned as soon as possible.
```
(Wording) -- I'd suggest:

> is planned immediately.

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+If a vulnerability affects the system directly, the update is planned as soon as possible.
+
+== Managed Server Maintenance Process
+All servers managed by VSHN are underlying a regular maintenance process.
```
(Wording) -- I'm not sure about the "underlying"; this might be Genglish. I'd suggest:

> VSHN executes a regular maintenance process for all managed services.

(Side note: It's not "servers", really, is it? It's not really "hosts" either, or is it?)

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+On that day the current patches and updates provided by the repositories of the running operation system are installed.
+
+Even though we at VSHN like to automate every possible process, the maintenance still has some manual steps.
+In theory the updates should not break running services, but since we are managing over 1000 servers with every update something could break.
```
Is the sheer number really the problem? This sounds like a fragile process. There must be a better reason why we do it manually.

Maybe we should say:

> Due to the complexity of incoming changes upgrades are supervised by human engineers.

Maybe merge the "complexity" aspect with the paragraph below. :thinking:

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+In theory the updates should not break running services, but since we are managing over 1000 servers with every update something could break.
+For this reason a VSHN engineer is supervising and running the updates.
+
+VSHN agrees with every customer, when their servers could be updated.
```
(Grammar) -- Don't use commas (only needed in German).

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+
+VSHN agrees with every customer, when their servers could be updated.
+We do have currently 4 maintenance windows.
+Tuesday morning, Tuesday afternoon starting 13:00, Tuesday evening starting 18:00, and finally Tuesday night starting 22:00.
```
I'd suggest to use bullet points, e.g.

> We have 4 maintenance windows: (Zurich local time)
> 
> * Tuesday morning
> * Tuesday afternoon starting 13:00
> * Tuesday evening starting 18:00
> * Tuesday night starting 22:00

(Side note: No need for "currently", I believe.)

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+VSHN agrees with every customer, when their servers could be updated.
+We do have currently 4 maintenance windows.
+Tuesday morning, Tuesday afternoon starting 13:00, Tuesday evening starting 18:00, and finally Tuesday night starting 22:00.
+Like that every customer can choose the window for their services regarding the criticality, time zones of end users, and other thoughts.
```
(Wording) -- I'd suggest:

> By that, we allow every customer to choose a service window that best fits their service needs [...]

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+Tuesday morning, Tuesday afternoon starting 13:00, Tuesday evening starting 18:00, and finally Tuesday night starting 22:00.
+Like that every customer can choose the window for their services regarding the criticality, time zones of end users, and other thoughts.
+
+Further there are services and servers, where customers chose only to update the servers once a month.
```
(Grammar) -- Don't use commas between partial sentences (German _Hauptsatz_ and _Nebensatz_), only the [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma) should be the exception.
(Grammar) -- Use temporal word at the beginning or the end of the sentence (not or rarely in the middle).
(Wording) -- Isn't "where" the Swiss "wo" (used for everything that start a dependent sentence)? Use _who_, _which_, or _that_ in doubt.
(Wording) -- I believe, "further" is a word motivated by German vocabulary. I'd suggest:

> In addition, there are services and hosts that some customers specifically choose to update once a month only.

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+Like that every customer can choose the window for their services regarding the criticality, time zones of end users, and other thoughts.
+
+Further there are services and servers, where customers chose only to update the servers once a month.
+This monthly maintenance is taking place on the first Tuesday of every month.
```
(Grammar) -- In doubt, avoid the *ing*-form. It's really just for things that happen *right now*.

> [...] takes place [...]

bittner commented on docs/modules/ROOT/pages/maintenance_process.adoc:
```diff
@@ -0,0 +1,78 @@
+Further there are services and servers, where customers chose only to update the servers once a month.
+This monthly maintenance is taking place on the first Tuesday of every month.
+
+For Software Services (SaaS) (for example Atlassian, nextcloud) and  Application Services (PaaS) (for example Tomcat) there are special maintenance and update procedures defined.
```
(Wording) -- Instead of "for example" I'd suggest using:

> e.g.

bittner reviewed (changes requested):
It's good to have this all described, but I'd love to have this more concise, if possible.

* Less sentences
* Shorter sentences
* More bulleted lists
* Maybe some illustrations en lieu de textual paragraphs

Also, we need to make sure we have native speakers reading over and approving the content (wording and grammar).

akosma committed:
Rephrased document

akosma commented:
@tobru I've gone through the text, please review and merge when ready! Thanks
